Index: xf86-video-qxl-0.1.4/src/qxl.h
===================================================================
--- xf86-video-qxl-0.1.4.orig/src/qxl.h
+++ xf86-video-qxl-0.1.4/src/qxl.h
@@ -276,6 +276,7 @@ struct _qxl_screen_t
 
 #ifdef XSERVER_LIBPCIACCESS
     struct pci_device *		pci;
+    struct pci_io_handle *	io;
 #else
     pciVideoPtr			pci;
     PCITAG			pci_tag;
@@ -634,7 +635,7 @@ void ioport_write(qxl_screen_t *qxl, uin
 #else
 static inline void ioport_write(qxl_screen_t *qxl, int port, int val)
 {
-    outb(qxl->io_base + port, val);
+    pci_io_write8(qxl->io, port, val);
 }
 #endif
 
Index: xf86-video-qxl-0.1.4/src/qxl_driver.c
===================================================================
--- xf86-video-qxl-0.1.4.orig/src/qxl_driver.c
+++ xf86-video-qxl-0.1.4/src/qxl_driver.c
@@ -220,6 +220,8 @@ unmap_memory_helper (qxl_screen_t *qxl)
 	pci_device_unmap_range (qxl->pci, qxl->vram, qxl->pci->regions[1].size);
     if (qxl->rom)
 	pci_device_unmap_range (qxl->pci, qxl->rom, qxl->pci->regions[2].size);
+    if (qxl->io)
+	pci_device_close_io (qxl->pci, qxl->io);
 #else
     if (qxl->ram)
 	xf86UnMapVidMem (scrnIndex, qxl->ram, (1 << qxl->pci->size[0]));
@@ -252,6 +254,9 @@ map_memory_helper (qxl_screen_t *qxl)
                           qxl->pci->regions[2].size, 0,
                           (void **)&qxl->rom);
     
+    qxl->io = pci_device_open_io(qxl->pci,
+                                qxl->pci->regions[3].base_addr,
+                                qxl->pci->regions[3].size);
     qxl->io_base = qxl->pci->regions[3].base_addr;
 #else
     qxl->ram = xf86MapPciMem (scrnIndex, VIDMEM_FRAMEBUFFER,
